libs/src/sdk/api/tracks/TracksApi.ts

```diff
@@ -144,13 +149,13 @@ export class TracksApi extends TracksApiWithoutStream {
     const metadataCid = await generateMetadataCidV1(updatedMetadata)
     const trackId = await this.generateTrackId()
     const response = await this.entityManager.manageEntity({
-      userId: artistId,
+      userId,
       entityType: EntityType.TRACK,
       entityId: trackId,
       action: Action.CREATE,
       metadata: JSON.stringify({
         cid: metadataCid.toString(),
-        data: updatedMetadata
+        data: snakecaseKeys(updatedMetadata)
```

Comment:
ik this isn't relevant for track metadata, but for future use, how would this work for something like `transcodedTrackCID` below where one of the components is all caps?
wonder if it would do `transcoded_track_CID` or `transcoded_track_c_i_d`?

Reply:
Such a good question!

Did some digging myself and found this:

`snakecase-keys` uses `snake-case` under the hood:
https://github.com/bendrucker/snakecase-keys/blob/master/index.js

Which uses `no-case` under the hood:
https://github.com/blakeembrey/change-case/blob/master/packages/snake-case/src/index.ts

Which splits while preserving all caps it seems:
https://github.com/blakeembrey/change-case/blob/040a079f007879cb0472ba4f7cc2e1d3185e90ba/packages/no-case/src/index.ts#L10

But has a default transform of `lower-case`
https://github.com/blakeembrey/change-case/blob/040a079f007879cb0472ba4f7cc2e1d3185e90ba/packages/no-case/src/index.ts#LL23C20-L23C20

Which uses `string.toLowerCase()`
https://github.com/blakeembrey/change-case/blob/040a079f007879cb0472ba4f7cc2e1d3185e90ba/packages/lower-case/src/index.ts#L54

So the result would be `transcoded_track_cid`